### DISCUSSION TITLE
Why css is duplicated?

### DISCUSSION THREAD

**Discussion by TolyaDeveloper** [mantinedev/mantine#3301] [Q&A] [answered]
### Code example
<img src="https://user-images.githubusercontent.com/49586997/211002808-12a92382-9edc-4323-af97-4b98f98650f9.png" width="450" height="400">

The only difference in css obviously is the margin property, but in resulting css we can see that each button has different css class and all properties duplicated (of course only margin differs):

<img src="https://user-images.githubusercontent.com/49586997/211003957-4755eb65-35b1-4d99-ad44-dfe190334616.png" width="450" height="550">

<img src="https://user-images.githubusercontent.com/49586997/211003774-80221a61-074e-4243-8e0d-f3cc4987961c.png" width="450" height="550">

<img src="https://user-images.githubusercontent.com/49586997/211004165-009166a2-4155-4f73-8921-c2648ec01fdb.png" width="450" height="550">

Isn't it better when separate classes will be created when custom css is written, and keep all the rest properties with the same class?, thus - no duplication.Btw, the same is happening with `sx` property

**Comment by rtivital** [comment, marked as answer]
It is designed this way to avoid conflicts among classes.
